extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/MutinyVertxVaultClient.java

```diff
@@ -268,129 +266,133 @@ public VaultWrapResult wrap(String token, long ttl, Object object) {
+        HttpRequest<Buffer> request = builder(path, token).method(HttpMethod.PUT);
+        return exec(request, body, resultClass);
     }
 
     protected <T> T put(String path, Object body, Class<T> resultClass) {
-        Request request = builder(path).put(requestBody(body)).build();
-        return exec(request, resultClass);
+        HttpRequest<Buffer> request = builder(path).method(HttpMethod.PUT);
+        return exec(request, body, resultClass);
     }
 
     protected <T> T get(String path, String token, Class<T> resultClass) {
-        Request request = builder(path, token).get().build();
+        HttpRequest<Buffer> request = builder(path, token).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected <T> T get(String path, Map<String, String> queryParams, Class<T> resultClass) {
-        final Request request = builder(path, queryParams).get().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected int head(String path) {
-        final Request request = builder(path).head().build();
+        final HttpRequest<Buffer> request = builder(path).method(HttpMethod.HEAD);
         return exec(request);
     }
 
     protected int head(String path, Map<String, String> queryParams) {
-        final Request request = builder(path, queryParams).head().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.HEAD);
         return exec(request);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass) {
-        return exec(request, resultClass, 200);
+    private <T> T exec(HttpRequest<Buffer> request, Class<T> resultClass) {
+        return exec(request, null, resultClass, 200);
+    }
+
+    private <T> T exec(HttpRequest<Buffer> request, int expectedCode) {
+        return exec(request, null, null, expectedCode);
     }
 
-    private <T> T exec(Request request, int expectedCode) {
-        return exec(request, null, expectedCode);
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass) {
+        return exec(request, body, resultClass, 200);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass, int expectedCode) {
-        try (Response response = client.newCall(request).execute()) {
-            if (response.code() != expectedCode) {
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass, int expectedCode) {
+        try {
+            HttpResponse<Buffer> response = body == null
+                    ? request.sendAndAwait()
+                    : request.sendJsonAndAwait(requestBody(body));
+            if (response.statusCode() != expectedCode) {
                 throwVaultException(response);
             }
-            String jsonBody = response.body().string();
+            String jsonBody = response.body().toString();
```

Comment:
Are you always expecting a json object? Can it be a Json array or something else sometimes?

Reply:
@cescoffier I have always seen json objects only in all endpoints I have had to work with. they usually start with this following common structure:
```
{
  "request_id": "9e923c23-3a45-ff0f-a3ea-5fffd94f1e2f",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": null,
  "wrap_info": null,
```

they are some exceptions on, but still returning objects.

Reply:
Ok, thanks! So we can simplify the code a bit by using the JSON Codec and retrieve JSON Object directly.